cfgmgr/teammgr.cpp

```diff
@@ -112,18 +115,48 @@ void TeamMgr::doTask(Consumer &consumer)
     }
 }
 
-
 void TeamMgr::cleanTeamProcesses()
 {
     SWSS_LOG_ENTER();
     SWSS_LOG_NOTICE("Cleaning up LAGs during shutdown...");
-    for (const auto& it: m_lagList)
+
+    struct timespec req = {.tv_sec = 0, .tv_nsec = 100000}; // 100 usec
+
+    for (const auto& alias: m_lagList)
     {
-        //This will call team -k kill -t <teamdevicename> which internally send SIGTERM 
-        removeLag(it);
+        pid_t pid;
+
+        if (!getLagPid(pid, alias))
+        {
+            continue;
+        }
+
+        SWSS_LOG_INFO("Send SIGTERM to port channel %s pid %d", alias.c_str(), pid);
+
+        if (kill(pid, SIGTERM) < 0)
+        {
+            SWSS_LOG_ERROR("Failed to send SIGTERM to port channel %s pid %d", alias.c_str(), pid);
+        }
     }
 
-    return;
+    for (const auto& alias: m_lagList)
+    {
+        pid_t pid;
+
+        if (!getLagPid(pid, alias))
+        {
+            continue;
+        }
+
+        SWSS_LOG_NOTICE("Waiting for port channel %s pid %d to stop...", alias.c_str(), pid);
+
+        while (kill(pid, 0) >= 0)
```

Comment:
@judyjoseph these warnings:
```bash
Aug 17 17:58:40.464776 sonic WARNING kernel: [ 3617.296910] PortChannel2 (unregistering): Failed to send options change via netlink (err -105)
Aug 17 17:58:40.464796 sonic WARNING kernel: [ 3617.297646] PortChannel62 (unregistering): Failed to send options change via netlink (err -105)
Aug 17 17:58:40.464803 sonic WARNING kernel: [ 3617.298464] PortChannel53 (unregistering): Failed to send options change via netlink (err -105)
```
can be fixed in two ways:
1. Adding sleep after `SIGTERM` is sent
2. Improving `teamd/kernel` behaviour to avoid race